### PR TITLE
New design of the candidate form 

### PR DESCRIPTION
**Pull request by veraivan**
## Description

Candidate, challenge and interview forms were restructured to be much more user friendly and have better usability. Also the page is adaptable for mobile devices which makes it easier to enter from a phone and do the management from there. 
Other changes were also made to the style guide and colors.

## Motivation and Context

This change improves usability for users, makes it much better understood and easier to use.
In addition to being able to login from your phone and easily add a candidate, it is extremely important that the systems are adaptable to mobile devices. That led me to implement these new changes in adding candidates.

## How Has This Been Tested?

Manual Testing
- Google Chrome, Responsive

### Do you have any tests?
- [ ] Yes
- [x] No

### CODE REVIEW

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html [hunk 1]
```diff
@@ -1,94 +1,139 @@
...
-        <label for="inputUniversity" class="col-sm-2 offset-sm-1 col-form-label text-label">University:</label>
-        <div class="col-sm-9">
+      <div class="form-group row mb-0" id="separator">
+        <label for="inputUniversity" class="col-sm-5 offset-sm-2 col-form-label text-label devices-adaptacion">University<span class="text-optional"> - Optional</span></label>
+      </div>
+      <div class="row">
+        <div class="col-sm-8 offset-sm-2 devices-adaptacion">
           <input type="text" formControlName="university" class="form-control input-field" id="inputUniversity"
-            placeholder="University">
+            placeholder="Input University">
```
Comment (mauri-medina) [inline front-end/src/app/candidate/data-candidate/data-candidate.component.html]: Please remove "Input" from the university input placeholder

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html [hunk 2]
```diff
@@ -1,94 +1,139 @@
...
-            <option value="Rejected">Rejected</option>
-          </select>
+      </div>
+      <div class="form-group row mb-0" id="separator">
+        <label for="filesUrl" class="col-sm-5 offset-sm-2 col-form-label text-label devices-adaptacion">Additional Files<span class="text-optional"> - Optional</span></label>
+      </div>
+      <div class="row">
+        <div class="col-sm-9 offset-sm-2 devices-adaptacion" id="filesUrl">
+          <input type="file" accept=".pdf, .doc" id="readfile2" hidden>
+          <button type="button" class="custom-button">Examine...</button>
```
Comment (mauri-medina) [inline front-end/src/app/candidate/data-candidate/data-candidate.component.html]: - Clicking the button does nothing, but I don't know if the button functionality will be added in a future ticket
-  I don't know if **"Examine..."** expresses the functionality of the button, I think that "**add file**" or "**select file**" better describes the action

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html [hunk 3]
```diff
@@ -1,94 +1,139 @@
...
-        <label for="inputEmail" class="col-sm-2 offset-sm-1 col-form-label text-label">Email:</label>
-        <div class="col-sm-9">
-          <input type="email" formControlName="email" class="form-control input-field" id="inputEmail" placeholder="Mail">
+      <div class="form-group row mb-0" id="separator">
+        <label for="inputLinkResume" class="col-sm-5 offset-sm-2 col-form-label text-label devices-adaptacion">Resume<span class="text-optional"> - Optional</span></label>
+      </div>
+      <div class="row">
+        <div class="col-sm-9 offset-sm-2 devices-adaptacion" id="inputLinkResume">
+          <input type="file" accept=".pdf, .doc" id="readfile1" hidden>
+          <button type="button" class="custom-button">Examine...</button>
```
Comment (mauri-medina) [inline front-end/src/app/candidate/data-candidate/data-candidate.component.html]: - Clicking the button does nothing, but I don't know if the button functionality will be added in a future ticket
- I don't know if "Examine..." expresses the functionality of the button, I think that "Select Resume" or "Select Resume" better describes the action

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html [hunk 4]
```diff
@@ -1,94 +1,139 @@
...
+      <div class="row">
+        <div class="col-sm-8 offset-sm-2 devices-adaptacion">
+          <textarea id="comments" class="form-control" formControlName="comments" rows="10" cols="12" placeholder="Input your comments"></textarea>
         </div>
       </div>
-      <div class="form-group row">
-        <div class="col-sm-12">
+      <div class="form-group row mt-4">
+        <div class="col-sm-11">
           <button type="submit" class="btn btn-pm">Save</button>
```
Comment (mauri-medina) [inline front-end/src/app/candidate/data-candidate/data-candidate.component.html]: When using the keyboard to navigate the form there is no indication that the save button have the focus

File: front-end/src/app/challenge/add-challenge/add-challenge.component.html [hunk 5]
```diff
@@ -1,63 +1,87 @@
...
         </div>
       </div>
     </div>
-
-    <div class="form-group row">
-      <div class="col-sm-12">
-        <button  type="submit"  (click)="onSubmit()" class="btn btn-pm">Save</button>
+    <div class="form-group row mt-4">
+      <div class="col-sm-10">
+        <button type="submit" class="btn btn-pm">Save</button>
```
Comment (mauri-medina) [inline front-end/src/app/challenge/add-challenge/add-challenge.component.html]: When using the keyboard to navigate the form there is no indication that the save button have the focus

File: front-end/src/app/interview/add-interview/add-interview.component.html [hunk 6]
```diff
@@ -1,71 +1,83 @@
...
+          </div>
+        </div>
         <!-- SAVE -->
-        <div class="form-group mt-4">
-          <button id="save-btn" type="submit" class="btn ok-btn font-weight-bold">Save</button>
-          <button id="cancel-btn" (click)="cancelInterview()" type="button" class="btn not-ok-btn font-weight-bold ml-3">Cancel</button>
+        <div class="row">
+          <div class="col-sm-11">
+            <button id="cancel-btn" (click)="cancelInterview()" type="button" class="btn not-ok-btn ml-3">Cancel</button>
+            <button id="save-btn" type="submit" class="btn ok-btn">Save</button>
```
Comment (mauri-medina) [inline front-end/src/app/interview/add-interview/add-interview.component.html]: When using the keyboard to navigate the form there is no indication that the save button have the focus

File: front-end/src/app/interview/add-interview/add-interview.component.html [hunk 7]
```diff
@@ -1,71 +1,83 @@
...
+            <div *ngIf="submitted && f.participants.errors">
+              <span class="text-danger" *ngIf="submitted && f.participants.errors?.required">Please add a participant</span>
             </div>
           </div>
-
-          <div *ngIf="participants.length > 0">
-            <ul><li *ngFor="let part of participants">{{part}}</li></ul>
+          <div class="col-sm-5 movil">
+            <button class="btn ok-btn-add" type="button" (click)="addParticipants()"
+              [disabled]="f.addParticipant.value == ''">Add Participants</button>
```
Comment (mauri-medina) [inline front-end/src/app/interview/add-interview/add-interview.component.html:63]: When using the keyboard to navigate the form there is no indication that the save button have the focus

**Review by mauri-medina — COMMENTED** [review]
Great redesign! there are some issues with the focus for buttons but otherwise it's looks great

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html (hunk 1 above)
Comment (veraivan) [reply front-end/src/app/candidate/data-candidate/data-candidate.component.html]: > Please remove "Input" from the university input placeholder

The word "Input" removed from the placeholder :+1:

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html (hunk 2 above)
Comment (veraivan) [reply front-end/src/app/candidate/data-candidate/data-candidate.component.html]: > * Clicking the button does nothing, but I don't know if the button functionality will be added in a future ticket
> * I don't know if **"Examine..."** expresses the functionality of the button, I think that "**add file**" or "**select file**" better describes the action

Regarding your first item. The functionality of that button, will be implemented in another ticket.  
It was changed from **"Examine"** to **"Select file"**.

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html (hunk 3 above)
Comment (veraivan) [reply front-end/src/app/candidate/data-candidate/data-candidate.component.html]: > * Clicking the button does nothing, but I don't know if the button functionality will be added in a future ticket
> * I don't know if "Examine..." expresses the functionality of the button, I think that "Select Resume" or "Select Resume" better describes the action

Regarding your first item. The functionality of that button, will be implemented in another ticket.  
It was changed from **"Examine"** to **"Select Resume"**.

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html (hunk 4 above)
Comment (veraivan) [reply front-end/src/app/candidate/data-candidate/data-candidate.component.html]: > When using the keyboard to navigate the form there is no indication that the save button have the focus

Thank you for that visualization. Added focus to the save button.

File: front-end/src/app/challenge/add-challenge/add-challenge.component.html (hunk 5 above)
Comment (veraivan) [reply front-end/src/app/challenge/add-challenge/add-challenge.component.html]: > When using the keyboard to navigate the form there is no indication that the save button have the focus

Thank you for that visualization. Added focus to the save button.

File: front-end/src/app/interview/add-interview/add-interview.component.html (hunk 7 above)
Comment (veraivan) [reply front-end/src/app/interview/add-interview/add-interview.component.html:63]: > When using the keyboard to navigate the form there is no indication that the save button have the focus

Thank you for that visualization. Added focus to the Add Participants button.

File: front-end/src/app/interview/add-interview/add-interview.component.html (hunk 6 above)
Comment (veraivan) [reply front-end/src/app/interview/add-interview/add-interview.component.html]: > When using the keyboard to navigate the form there is no indication that the save button have the focus

Thank you for that visualization. Focus has been added to the save and cancel buttons.

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html [hunk 8]
```diff
@@ -1,94 +1,139 @@
...
-          <span class="text-danger" *ngIf="submitted && f.name.errors?.pattern">Name have to be only letters</span>
-          <span class="text-danger" *ngIf="submitted && f.name.errors?.maxlength">Name must have less than 100
-          characters</span>
+    <div class="col-sm-9">
+      <div class="form-group mb-0 mt-3 row">
+        <label for="inputName" class="col-sm-2 offset-sm-2 col-form-label text-label devices-adaptacion">Name</label>
+      </div>
+      <div class="row">
+        <div class="col-sm-5 offset-sm-2 devices-adaptacion">
+          <input type="text" formControlName="name" class="form-control input-field" id="inputName" placeholder="Input Name">
```
Comment (mauri-medina) [inline front-end/src/app/candidate/data-candidate/data-candidate.component.html]: Please remove the word "Input" from the input placeholder

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html [hunk 9]
```diff
@@ -1,94 +1,139 @@
...
-        <div class="col-sm-3">
-          <div class=" date-picker-width">
+      </div>
+      <div class="form-group row mb-0" id="separator">
+        <label for="inputLastName" class="col-sm-2 offset-sm-2 col-form-label text-label devices-adaptacion">Last Name</label>
+      </div>
+      <div class="row">
+        <div class="col-sm-5 offset-sm-2 devices-adaptacion">
+          <input type="text" formControlName="lastName" class="form-control input-field" id="inputLastName"
+            placeholder="Input Last Name">
```
Comment (mauri-medina) [inline front-end/src/app/candidate/data-candidate/data-candidate.component.html]: Please remove the word "Input" from the input placeholder

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html [hunk 10]
```diff
@@ -1,94 +1,139 @@
...
+          <span class="text-eror" *ngIf="submitted && f.name.errors?.pattern">Last Name have to be only letters</span>
+          <span class="text-eror" *ngIf="submitted && f.name.errors?.maxlength">Last Name must have less than 100characters</span>
+        </div>
+      </div>
+      <div class="form-group row mb-0" id="separator">
+          <label for="inputEmail" class="col-sm-3 offset-sm-2 text-label devices-adaptacion">Email<span class="text-optional"> - Optional</span></label>
+      </div> 
+      <div class="row">
+        <div class="col-sm-6 offset-sm-2 devices-adaptacion">
+          <input type="email" formControlName="email" class="form-control input-field" id="inputEmail" placeholder="Input Email">
```
Comment (mauri-medina) [inline front-end/src/app/candidate/data-candidate/data-candidate.component.html]: Please remove the word "Input" from the input placeholder

File: front-end/src/app/challenge/add-challenge/add-challenge.component.html [hunk 11]
```diff
@@ -1,63 +1,87 @@
...
-      <label class="margin-left text" for="comment">Comment</label>
-      <textarea class="form-control margin-left comment-width" formControlName="comment" rows="3"
+    <div class="from-group row mb-0" id="separator">
+      <div class="col-sm-3 offset-sm-3">
+        <label class="text-label">Reviewer</label>
+      </div>
+    </div>
+    <div class="row">
+      <div class="col-sm-5 offset-sm-3">
+        <input type="text-label" formControlName="reviewer" placeholder="Input Add a Reviewer" class="form-control"
```
Comment (mauri-medina) [inline front-end/src/app/challenge/add-challenge/add-challenge.component.html]: Please remove the word "Input" from the input placeholder

File: front-end/src/app/challenge/add-challenge/add-challenge.component.html [hunk 12]
```diff
@@ -1,63 +1,87 @@
...
+      </div>
+    </div>
+    <div class="form-group row mb-0" id="separator">
+      <div class="col-sm-3 offset-sm-3">
+        <label class="text-label">Link Challenge</label>
+      </div>
+    </div>
+    <div class="row">
+      <div class="col-sm-5 offset-sm-3">
+        <input type="text-label" formControlName="linkChallenge" placeholder="Input Add a link" class="form-control"
```
Comment (mauri-medina) [inline front-end/src/app/challenge/add-challenge/add-challenge.component.html]: Please remove the word "Input" from the input placeholder

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html [hunk 13]
```diff
@@ -1,94 +1,139 @@
...
-      <div class="form-group row">
-        <label for="comments" class="col-sm-2 offset-sm-1 col-form-label text-label">Comments:</label>
-        <div class="col-sm-9">
-          <textarea id="comments" class="form-control" formControlName="comments"></textarea>
+      <div class="form-group row mb-0" id="separator">
+        <label for="comments" class="col-sm-4 offset-sm-2 col-form-label text-label devices-adaptacion">Comments</label>
+      </div>
+      <div class="row">
+        <div class="col-sm-8 offset-sm-2 devices-adaptacion">
+          <textarea id="comments" class="form-control" formControlName="comments" rows="10" cols="12" placeholder="Input your comments"></textarea>
```
Comment (mauri-medina) [inline front-end/src/app/candidate/data-candidate/data-candidate.component.html]: I think a better placeholder for the text area could be "Write your comments here", like the one in the text area in the challenge section

File: front-end/src/app/interview/edit-interview/edit-interview.component.html [hunk 14]
```diff
@@ -29,37 +21,48 @@
...
-        <!-- ADD PARTICIPANT -->
-        <div class="form-group">
-          <label for="participants" class="font-weight-bold">Participants</label>
-
-          <div class="row">
-            <div class="col-12 col-md-9 mb-3">
-              <input id="participant" type="text" formControlName="addParticipant" placeholder="Add Participants"
+        <div class="row">
+          <div class="col-sm-5 offset-sm-2">
+            <input id="participant" type="text" formControlName="addParticipant" placeholder="Add Participants"
```
Comment (mauri-medina) [inline front-end/src/app/interview/edit-interview/edit-interview.component.html:65]: - When i hover the "Add Participants" button with the mouse it looks like something in the page resize, im testing this in chrome 

- To me, the trash can icon looks too big and adds an unwanted vertical space, but this is a subjective point of view if you think it's better like it is now it is fine too.o 👍 

image

File: front-end/src/app/interview/list-interview/list-interview.component.html [hunk 15]
```diff
@@ -1,68 +1,69 @@
...
-        <button class="btn interview-btn-pm" id="buttonfinish" (click)="onOpenDialog($event); isFinish = false;"
-          [disabled]="isFinish">Finish all interviews</button>
-        <ejs-dialog id='dialog' #ejDialog header='Are you sure ...? '
-          content='If you finish the process you wont be able to add or edit' [target]='targetElement' width='300px'
-          [buttons]='buttons' [visible]='hidden'>
-        </ejs-dialog>
-      </div>
+    <div class="col-sm-12">
+      <button class="btn interview-btn-pm" id="buttonfinish" (click)="onOpenDialog($event); isFinish = false;"
+        [disabled]="isFinish">Finish all interviews</button>
```
Comment (mauri-medina) [inline front-end/src/app/interview/list-interview/list-interview.component.html:62]: - When i hover the "Finish all interviews" button with the mouse it looks like something in the page resize, im testing this in chrome
- To me, the pencil icon looks too big and adds an unwanted vertical space, but this is a subjective point of view if you think it's better like it is now it is fine too 👍

image

**Review by mauri-medina — COMMENTED** [review]
Thanks for the changes! It's  looking great already. There are some minor things with some buttons like the "Save" buttons when you hover them with the mouse, it cause a resizing or something like that, if you could please check this so we can add your redesign to the project

**Comment by luifer-villalba** [comment]
I don't know if this is a known issue. 
Let's say I'm creating/updating a profile and when I clicked Enter in any field. 


In this case, I clicked Enter in the email input -> the image disappeared.

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html (hunk 8 above)
Comment (veraivan) [reply front-end/src/app/candidate/data-candidate/data-candidate.component.html]: > Please remove the word "Input" from the input placeholder

The word "Input" removed from the Name placeholder.

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html (hunk 9 above)
Comment (veraivan) [reply front-end/src/app/candidate/data-candidate/data-candidate.component.html]: > Please remove the word "Input" from the input placeholder

The word "Input" removed from the Last Name placeholder.

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html (hunk 10 above)
Comment (veraivan) [reply front-end/src/app/candidate/data-candidate/data-candidate.component.html]: > Please remove the word "Input" from the input placeholder

The word "Input" removed from the Email placeholder.

File: front-end/src/app/challenge/add-challenge/add-challenge.component.html (hunk 11 above)
Comment (veraivan) [reply front-end/src/app/challenge/add-challenge/add-challenge.component.html]: > Please remove the word "Input" from the input placeholder

The word "Input" removed from the Add a Reviewer placeholder.

File: front-end/src/app/challenge/add-challenge/add-challenge.component.html (hunk 12 above)
Comment (veraivan) [reply front-end/src/app/challenge/add-challenge/add-challenge.component.html]: > Please remove the word "Input" from the input placeholder

The word "Input" removed from the Add a link placeholder.

File: front-end/src/app/candidate/data-candidate/data-candidate.component.html (hunk 13 above)
Comment (veraivan) [reply front-end/src/app/candidate/data-candidate/data-candidate.component.html]: > I think a better placeholder for the text area could be "Write your comments here", like the one in the text area in the challenge section

Change made :+1:

File: front-end/src/app/interview/edit-interview/edit-interview.component.html (hunk 14 above)
Comment (veraivan) [reply front-end/src/app/interview/edit-interview/edit-interview.component.html]: > * To me, the trash can icon looks too big and adds an unwanted vertical space, but this is a subjective point of view if you think it's better like it is now it is fine too.o

Thank you for your suggestions :ok_hand:. I made the corresponding improvements to the buttons, in addition to reducing the size of the garbage icon.

File: front-end/src/app/interview/list-interview/list-interview.component.html (hunk 15 above)
Comment (veraivan) [reply front-end/src/app/interview/list-interview/list-interview.component.html:62]: > * When i hover the "Finish all interviews" button with the mouse it looks like something in the page resize, im testing this in chrome
> * To me, the pencil icon looks too big and adds an unwanted vertical space, but this is a subjective point of view if you think it's better like it is now it is fine too +1
> 
> image

Thank you for your suggestions :ok_hand:
I made the corresponding improvements to the buttons, in addition to reducing the size of the pencil icon.

File: front-end/src/app/interview/edit-interview/edit-interview.component.html (hunk 14 above)
Comment (veraivan) [reply front-end/src/app/interview/edit-interview/edit-interview.component.html:65]: > * When i hover the "Add Participants" button with the mouse it looks like something in the page resize, im testing this in chrome
> * To me, the trash can icon looks too big and adds an unwanted vertical space, but this is a subjective point of view if you think it's better like it is now it is fine too.o +1
> 
> image

Thank you for your suggestions :ok_hand:
I made the corresponding improvements to the buttons, in addition to reducing the size of the garbage icon.

**Comment by veraivan** [comment]
> I don't know if this is a known issue.
> Let's say I'm creating/updating a profile and when I clicked Enter in any field.
> 
> 
> In this case, I clicked Enter in the email input -> the image disappeared.
>

**Comment by veraivan** [comment]
> I don't know if this is a known issue.
> Let's say I'm creating/updating a profile and when I clicked Enter in any field.
> 
> 
> In this case, I clicked Enter in the email input -> the image disappeared.
>

**Comment by veraivan** [comment]
> I don't know if this is a known issue.
> Let's say I'm creating/updating a profile and when I clicked Enter in any field.
> 
> 
> In this case, I clicked Enter in the email input -> the image disappeared.
> 

Error solved :+1:

File: front-end/src/app/candidate/list-candidate/list-candidate.component.css [hunk 16]
```diff
@@ -18,15 +18,15 @@ ngb-pagination ::ng-deep .page-item.active .page-link {
   border-color: #3498db;
 }
 
-
 .btn-outline-success:hover{
   color: white;
   background-color: #3498db;
   border-color: #3498db;
+<<<<<<< HEAD
```
Comment (mauri-medina) [inline front-end/src/app/candidate/list-candidate/list-candidate.component.css]: Please remove these lines left by the merge

File: front-end/src/app/candidate/list-candidate/list-candidate.component.css [hunk 17]
```diff
@@ -18,15 +18,15 @@ ngb-pagination ::ng-deep .page-item.active .page-link {
...
+=======
 }
 
 #filter-input, #select-sortby {
   width: 10rem;
-}
-
-/*@import "../node_modules/@syncfusion/ej2/material.css";*/
+>>>>>>> ed4d77746c079c284f444231d8257262cd2caaf9
```
Comment (mauri-medina) [inline front-end/src/app/candidate/list-candidate/list-candidate.component.css]: Please remove these lines left by the merge

**Review by mauri-medina — COMMENTED** [review]
Thanks for the changes, please remove these lines left by the merge and that's it for me, Thanks for keeping your branch updated. Great work 👍

**Review by mauri-medina — APPROVED** [review]

**Review by luifer-villalba — APPROVED** [review]
Thanks for your collaboration. The candidate form looks neat and clean.

**Pull request merged**